File: 卡码/代码随想录/1/09-动态规划/Go/02-背包问题/02-完全背包/LeetCode_0139.go
// 139. 单词拆分
package main

import (
	"slices"
)

func wordBreak(s string, wordDict []string) bool {
	dp := make([]bool, len(s)+1)
	for i := 1; i <= len(s); i++ {
		dp[i] = false
	}
	dp[0] = true
	for i := 1; i <= len(s); i++ {
		for j := 0; j < i; j++ {
			if slices.Contains(wordDict, s[j:i]) && dp[j] == true {
				dp[i] = true
			}
		}
	}
	return dp[len(s)]
}

File: 卡码/代码随想录/1/09-动态规划/Js/02-背包问题/02-完全背包/LeetCode_0139.js
//139. 单词拆分
const wordBreak = function(s, wordDict) {
	const dp=new Array(s.length+1).fill(false);
	dp[0]=true;
	for(let i=1;i<=s.length;i++){
		for(let j=0;j<i;j++){
			if(wordDict.includes(s.slice(j,i))&&dp[j]) dp[i]=true;
		}
	}
	return dp[s.length];
};